hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/HdfsMesosConstraints.java

```diff
@@ -0,0 +1,113 @@
+package org.apache.mesos.hdfs.scheduler;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.mesos.Protos.Attribute;
+import org.apache.mesos.Protos.Offer;
+import org.apache.mesos.Protos.Value.Range;
+import org.apache.mesos.hdfs.config.HdfsFrameworkConfig;
+
+/**
+ * HDFS Mesos offer constraints checker class implementation.
+ */
+public class HdfsMesosConstraints {
+  
+  private final Log log = LogFactory.getLog(HdfsMesosConstraints.class);
+  private final HdfsFrameworkConfig config;
+
+  public HdfsMesosConstraints(HdfsFrameworkConfig config) {
+    this.config = config;
+  }
+
+  public boolean constraintsAllow(Offer offer) {
+    List<Attribute> attributes = offer.getAttributesList();
+    
+    Map<String, String> constraints = config.getMesosSlaveConstraints();
+    Set<Map.Entry<String, String>> constraintSet = constraints.entrySet();
+    
+    for (Map.Entry<String, String> constraintEntry : constraintSet) {
+      boolean found = false;
+      String constraintName = constraintEntry.getKey();
+      String constraintValue = constraintEntry.getValue();
+      
+      for (Attribute attribute : attributes) {
+        if (attribute.getName().equals(constraintName)) {
+          switch (attribute.getType()) {
+          case RANGES:
+            if (attribute.hasRanges()) {
+              try {
+                Long range = Long.parseLong(constraintValue);
+                for (Range r : attribute.getRanges().getRangeList()) {
+                  if ((!r.hasBegin() || range >= r.getBegin())
+                      && (!r.hasEnd() || range <= r.getEnd())) {
+                    found = true;
+                    break;
+                  }
+                }
+              } catch (NumberFormatException e) {
+                // Offer attribute value is not castble to number.
+                String msg = "Constraint value " + constraintValue + 
+                    " is not of type range for offer attribute " + constraintName; 
+                log.warn(msg, e);
+              }
+            } 
+            break;
+          case SCALAR:
+            if (attribute.hasScalar()) {
+              try {
+                if (attribute.getScalar().getValue() >= Double
+                    .parseDouble(constraintValue)) {
+                  found = true;
+                }
+              } catch (NumberFormatException e) {                
+                // Offer attribute value is not castble to scalar. 
+                String msg = "Constraint value \"" + constraintValue + 
+                "\" is not of type scalar for offer attribute " + constraintName;
+                log.warn(msg, e);
+              }
```

Comment:
I will still put this as warning, because it is not an error if given constraint value is not of type of offered attribute. 

For example, there might two slave machines A and B. If A has attribute "ids" of type range and machine B has attribute "ids" of type text. In user has specified the constraint considering value of type text, then for offers from slave A, code may catch here. In that case, it should just be a warning, not an error. 

This might be rare condition, but not stoppable.